csrc/quantization/cutlass_w8a8/scaled_mm_dq_bias_entry.cu

```diff
@@ -0,0 +1,76 @@
+#include <cudaTypedefs.h>
+
+#include <cassert>
+#include <c10/cuda/CUDAGuard.h>
+#include <torch/all.h>
+
+// void cutlass_scaled_mm_dq_sm75(torch::Tensor& c, torch::Tensor const& a,
+//                                torch::Tensor const& b,
+//                                torch::Tensor const& a_scales,
+//                                torch::Tensor const& b_scales);
+
+// void cutlass_scaled_mm_dq_sm80(torch::Tensor& c, torch::Tensor const& a,
+//                                torch::Tensor const& b,
+//                                torch::Tensor const& a_scales,
+//                                torch::Tensor const& b_scales);
+
+// void cutlass_scaled_mm_dq_sm89(torch::Tensor& c, torch::Tensor const& a,
+//                                torch::Tensor const& b,
+//                                torch::Tensor const& a_scales,
+//  torch::Tensor const& b_scales);
+
+void cutlass_scaled_mm_dq_bias_sm90(torch::Tensor& c, torch::Tensor const& a,
+                                    torch::Tensor const& b,
+                                    torch::Tensor const& a_scales,
+                                    torch::Tensor const& b_scales,
+                                    torch::Tensor const& bias);
+
+void cutlass_scaled_mm_dq_bias(torch::Tensor& c, torch::Tensor const& a,
+                               torch::Tensor const& b,
+                               torch::Tensor const& a_scales,
+                               torch::Tensor const& b_scales,
+                               torch::Tensor const& bias) {
+  int32_t major_capability;
+  int32_t minor_capability;
+  cudaDeviceGetAttribute(&major_capability, cudaDevAttrComputeCapabilityMajor,
+                         0);
+  cudaDeviceGetAttribute(&minor_capability, cudaDevAttrComputeCapabilityMinor,
+                         0);
+  int32_t version_num = major_capability * 10 + minor_capability;
+
+  // Checks for conformality
+  TORCH_CHECK(a.dim() == 2 && b.dim() == 2 && c.dim() == 2);
+  TORCH_CHECK(c.size(0) == a.size(0) && a.size(1) == b.size(0) &&
+              b.size(1) == c.size(1));
+  TORCH_CHECK(a_scales.numel() == 1 || a_scales.numel() == a.size(0));
+  TORCH_CHECK(b_scales.numel() == 1 || b_scales.numel() == b.size(1));
+
+  // Check for strides and alignment
+  TORCH_CHECK(a.stride(1) == 1 && c.stride(1) == 1);  // Row-major
+  TORCH_CHECK(b.stride(0) == 1);                      // Column-major
+  TORCH_CHECK(c.stride(0) % 16 == 0 &&
+              b.stride(1) % 16 == 0);  // 16 Byte Alignment
+  TORCH_CHECK(a_scales.is_contiguous() && b_scales.is_contiguous());
+
+  // TODO: add checks for bias shape and alignment
+
+  at::cuda::OptionalCUDAGuard const device_guard(device_of(a));
+
+  if (version_num >= 90) {
+    // Hopper
+    cutlass_scaled_mm_dq_bias_sm90(c, a, b, a_scales, b_scales, bias);
+  } else {
+    assert(0 && "cutlass_scaled_mm_dq_bias only supports Hopper for now");
```

Comment:
Can you please support Ada as well for FP8? Your test seems like it would be running for Ada